README.md
# NAME

Acme::TestDist::Cpp::EUMM::EUCppGuess - Example C++ distribution with MakeMaker and CppGuess to test the tool chain

<div>
    <a href="https://travis-ci.org/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess"><img src="https://travis-ci.org/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess.png" alt="Acme-TestDist-Cpp-EUMM-EUCppGuess"></a>
    <a href='https://coveralls.io/r/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess?branch=master'><img src='https://coveralls.io/repos/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess/badge.png?branch=master' alt='Coverage Status' /></a>
    <a href='http://cpants.cpanauthors.org/dist/Acme-TestDist-Cpp-EUMM-EUCppGuess'><img src='http://cpants.cpanauthors.org/dist/Acme-TestDist-Cpp-EUMM-EUCppGuess.png' alt='Kwalitee Score' /></a>
    <a href="http://badge.fury.io/pl/Acme-TestDist-Cpp-EUMM-EUCppGuess"><img src="https://badge.fury.io/pl/Acme-TestDist-Cpp-EUMM-EUCppGuess.svg" alt="CPAN version" height="18"></a>
</div>

# SYNOPSIS

    use Acme::TestDist::Cpp::EUMM::EUCppGuess qw(return_one);

    return_one(); # returns 1

# DESCRIPTION

This distribution is useful for

- being a template to start your own CPAN XS/C++ distribution
- test the tool chain of an individual environment
- test the tool chains of CPAN infrastructure

# EXPORT

None by default, but the following functions are available:

## return\_one()

Returns the number 1. Useful for test.

## returnOne()

Internal use.

# VERSIONING

After reaching near perfect quality version numbers will be in the form
year.month.minor, i.e. YYYY.MM.99.

# SEE ALSO

`ExtUtils::CppGuess`
`Extutils::MakeMaker`

# SOURCE REPOSITORY

[http://github.com/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess](http://github.com/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess)

# AUTHOR

Helmut Wollmersdorfer &lt;helmut.wollmersdorfer@gmail.com>

<div>
    <a href='http://cpants.cpanauthors.org/author/wollmers'><img src='http://cpants.cpanauthors.org/author/wollmers.png' alt='Kwalitee Score' /></a>
</div>

# COPYRIGHT AND LICENSE

Copyright (C) 2015, 2016 Helmut Wollmersdorfer

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.


Changes
Revision history for Perl extension Acme::TestDist::Cpp::EUMM::EUCppGuess.

0.01  Fri Jan  1 01:33:00 2016
    - add MANIFEST

0.01  Fri Jan  1 00:40:00 2016
	- original version



Makefile.PL
use 5.006;
use ExtUtils::MakeMaker;
# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.

use ExtUtils::CppGuess;

my $guess = ExtUtils::CppGuess->new;

WriteMakefile(
    NAME              => 'Acme::TestDist::Cpp::EUMM::EUCppGuess',
    AUTHOR			  => 'Helmut Wollmersdorfer <helmut.wollmersdorfer@gmail.com>',
    ABSTRACT_FROM     => 'lib/Acme/TestDist/Cpp/EUMM/EUCppGuess.pm',
    VERSION_FROM      => 'lib/Acme/TestDist/Cpp/EUMM/EUCppGuess.pm', # finds $VERSION
    MIN_PERL_VERSION  => '5.6.0',
    PREREQ_PM         => {}, # e.g., Module::Name => 1.1
    dist	 => {
	    PREOP	=> 'pod2text lib/Acme/TestDist/Cpp/EUMM/EUCppGuess.pm | tee README >$(DISTVNAME)/README; pod2markdown lib/Acme/TestDist/Cpp/EUMM/EUCppGuess.pm > README.md; chmod -R u=rwX,go=rX . ;',
	    COMPRESS	=> 'gzip -9v',
	    SUFFIX	=> '.gz',
	},
    $guess->makemaker_options,

    TEST_REQUIRES		=> {
            'Test::More' => '0.88',
    },
    META_MERGE => {
        'meta-spec' => { version => 2 },
        license  => [ 'perl_5' ],
        prereqs => {
  			runtime => {
    			requires => {
      				'perl'          => '5.006',
    			},
    			recommends => {
    			},
    			suggests => {
    			},
  			},
 			build => {
    			requires => {
    			    'ExtUtils::MakeMaker' => '0',
      				'ExtUtils::CppGuess' => '0.11',
    			},
  			},
  			configure => {
         		requires => {
            		'ExtUtils::MakeMaker' => '0',
            		'ExtUtils::CppGuess' => '0.11',
         		}
      		},
  			test => {
    			requires => {
            		'Test::More' 	  => '0.88',
    			},
  			}
		},
        resources => {
            repository => {
                type => 'git',
                url  => 'https://github.com/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess.git',
                web  => 'https://github.com/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess',
            },
            homepage => 'https://github.com/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess',
            bugtracker => {
                web => 'https://github.com/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess/issues',
            },
        },
     },
);



lib/Acme/TestDist/Cpp/EUMM/EUCppGuess.pm
package Acme::TestDist::Cpp::EUMM::EUCppGuess;

use 5.006;
use strict;
use warnings;

require Exporter;
use AutoLoader;

our @ISA = qw(Exporter);
our %EXPORT_TAGS = ( 'all' => [ qw( return_one ) ] );
our @EXPORT_OK = ( @{ $EXPORT_TAGS{'all'} } );
our @EXPORT = qw( );

our $VERSION = '0.02';

require XSLoader;
XSLoader::load('Acme::TestDist::Cpp::EUMM::EUCppGuess', $VERSION);

sub return_one {
  returnOne();
}



1;
__END__

=head1 NAME

Acme::TestDist::Cpp::EUMM::EUCppGuess - A c++ distribution for testing the toolchain

=begin html

<a href="https://travis-ci.org/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess"><img src="https://travis-ci.org/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess.png" alt="Acme-TestDist-Cpp-EUMM-EUCppGuess"></a>
<a href='https://coveralls.io/r/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess?branch=master'><img src='https://coveralls.io/repos/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess/badge.png?branch=master' alt='Coverage Status' /></a>
<a href='http://cpants.cpanauthors.org/dist/Acme-TestDist-Cpp-EUMM-EUCppGuess'><img src='http://cpants.cpanauthors.org/dist/Acme-TestDist-Cpp-EUMM-EUCppGuess.png' alt='Kwalitee Score' /></a>
<a href="http://badge.fury.io/pl/Acme-TestDist-Cpp-EUMM-EUCppGuess"><img src="https://badge.fury.io/pl/Acme-TestDist-Cpp-EUMM-EUCppGuess.svg" alt="CPAN version" height="18"></a>

=end html

=head1 SYNOPSIS

    use Acme::TestDist::Cpp::EUMM::EUCppGuess qw(return_one);

    return_one(); # returns 1

=head1 DESCRIPTION

ImgSeek (http://www.imgseek.net/) is an implementation of Haar wavelet
decomposition techniques to find similar pictures in a library. This
module is port of the ImgSeek library to Perl's XS. It can deal with
image objects produced by the C<Imager>, C<Image::Imlib2> and C<GD> libraries.

=head1 EXPORT

None by default, but the following functions are available:

=head2 return_one()

Returns the number 1.

=head1 SEE ALSO



=head1 SOURCE REPOSITORY

L<http://github.com/wollmers/Acme-TestDist-Cpp-EUMM-EUCppGuess>

=head1 MAINTAINER

Helmut Wollmersdorfer E<lt>helmut.wollmersdorfer@gmail.comE<gt>

=begin html

<a href='http://cpants.cpanauthors.org/author/wollmers'><img src='http://cpants.cpanauthors.org/author/wollmers.png' alt='Kwalitee Score' /></a>

=end html

=head1 AUTHOR

Helmut Wollmersdorfer, E<lt>helmut.wollmersdorfer@gmail.comE<gt>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2015, 2016 Helmut Wollmersdorfer

This library is free software; as it is a derivative work of imgseek,
this library is distributed under the same terms (GPL) as imgseek.

=cut
